dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/ProgramRuleAction.java

```diff
@@ -251,6 +257,42 @@ public ProgramRuleAction(
     this.programRuleActionEvaluationEnvironments = environments;
   }
 
+  public ProgramRuleAction(
```

Comment:
I don't think this constructor is needed.
I know that you are following a pattern in this class, but these constructors are only used in tests and we can just use the setter if we will need to set the priority.
Hibernate and the json serializer/deserializer work with the default constructor without arguments and the setters.